src/suspense.js

```diff
@@ -0,0 +1,52 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+export class Suspense extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {};
+	}
+
+	__s(e) {
+		if (typeof e.then == 'function') {
+			this.setState({ l: 1 });
+			const cb = () => { this.setState({ l: 0 }); };
+
+			// Suspense ignores errors thrown in Promises as this should be handled by user land code
+			e.then(cb, cb);
+		}
+		else {
+			throw e;
+		}
+	}
+
+	render(props, state) {
+		return state.l ? props.fallback : props.children;
+	}
+}
+
+export function lazy(loader) {
+	let prom;
+	let component;
+	let error;
+	return function L(props) {
```

Comment:
@marvinhagemeister does this function need a `displayName` to show up in devtools correctly? Similarly to `memo` and `forwardRef`?

Reply:
Good catch! Yes it does need the `displayName` property to be able to distinguish it from other components 👍

Reply:
Added a `displayName` that is set once the lazy loaded component is loaded. Will check whether this works once I got a demo together